bundle/manifests/elasticsearch-operator-alerts_monitoring.coreos.com_v1_prometheusrule.yaml

```diff
@@ -0,0 +1,24 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  labels:
+    name: elasticsearch-operator
+  name: elasticsearch-operator-alerts
+spec:
+  groups:
+  - name: elasticsearch_operator.alerts
+    rules:
+    - alert: ElasticsearchOperatorRemovalWarning
+      annotations:
+        message: Elasticsearch Operator deprecated and will be removed from the catalog
+          on November 2025.
+        summary: In OCP 4.16 and forward, Elasticsearch is only supported for ServiceMesh,
+          Tracing and Kiali, and this operator will be removed from the OpenShift
+          Operator Catalog in November 2025. Elasticsearch is no longer supported
+          for log storage. Kibana is no longer supported in OCP 4.16 and forward.
+      expr: |
+        last_over_time((cluster_version{version=~"4.1(6|7|8).*", type="current"} and on(prometheus) csv_succeeded{exported_namespace="openshift-operators-redhat", name=~"elasticsearch-operator.*", version=~"5.8.*"})[5m:]) > 0
```

Comment:
This should prevent many-to-many match errors by aggregating away unnecessary labels. The only preserved label is  the operator `version` which could be useful to track.
```suggestion
        last_over_time((max by(version) (csv_succeeded{exported_namespace="openshift-operators-redhat", name=~"elasticsearch-operator.*", version=~"5.8.*"}) * on() group_left() group(cluster_version{version=~"4.1[6-9].*", type="current"}))[5m:]) > 0
```
Tested with a 4.16 cluster and it returns a result

![image](https://github.com/user-attachments/assets/f1d51550-f501-4fab-87a0-f6ab0f406c15)